x-pack/test/ui_capabilities/security_and_spaces/scenarios.ts

```diff
@@ -0,0 +1,520 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Space, User } from '../common/types';
```

Comment:
It might be worthwhile to add a sentence or two here explaining our motivation for structuring the scenarios this way (e.g., enables exhaustive switches)